94: Add Tabs and TabsItem components

Caleb-Ellis opened the pull request:
# Done
- Added [Tabs](https://docs.vanillaframework.io/en/patterns/tabs) component and TabsItem subcomponent
- Added stories/knobs to Storybook
- Added snapshot tests

# QA
- Pull code
- Run `yarn lint` and `yarn test` to ensure there are no linting or testing errors
- Run `./run serve --watch` and navigate to http://localhost:8102/
- Verify the Tabs component in Storybook matches the [Vanilla docs](https://docs.vanillaframework.io/en/patterns/tabs)

# Fixes
Fixes #94

Commit:
94: Add Tabs component

webteam-app commented:
Starting demo at: http://vanilla-framework-react-pr-97.run.demo.haus/

anthonydillon commented on src/lib/components/Tabs/TabsItem.js:
```diff
@@ -0,0 +1,55 @@
+        <a
+          href={href}
+          className="p-tabs__link"
+          tabIndex={isSelected ? 0 : -1}
```
Can you remove this tabIndex as -1 stops a user from tabbing to it.

anthonydillon reviewed (changes requested):
One small comment about the use of tabIndex

Caleb-Ellis commented:
@anthonydillon I've changed tabIndex to be index + 1 so you can still tab through in order, and added an onKeyPress func so you can navigate the tabs via keyboard

anthonydillon reviewed (approved):
Code looks good

webteam-app commented:
Starting demo at: http://vanilla-framework-react-pr-97.run.demo.haus/
